### PR TITLE
Updates for Readme

### PR DESCRIPTION
**Pull request by h-unterp**
1. Corrected english grammatical errors. 
2. Formatted code
3. Added some explanations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
 
-This **StateNotifierProvider** must have two things to ensure everything works correctly, it must have a **load method** and it must have a state that has the list of the elements, an error, and a variable that hold the next page that the load function will use.
+This `StateNotifierProvider` must implement two methods to ensure everything works correctly, it must have a `load method`, and a `nextPageKeyBuilder` method, these will be explained below.
 
-The **riverpod_infinite_scroll** help us to ensure that the **StateNotifier** we are using respect this constraints with two classes:
+`riverpod_infinite_scroll` helps us to ensure that our `StateNotifier` will respect these constraints with the choice of two classes:
 
- - **PagedNotifierMixin** - a mixin that ensure the **StateNotifier** will implement the right **load** method
- - **PagedState** - a state that has all the properties that **riverpod_infinite_scroll** need
+You can either use the simple:
```
Comment (ftognetto) [inline README.md:24]: In this section I should have explained better. I will commit an edit here

**Review by ftognetto — COMMENTED** [review]
I would like to thank you very much for taking the time to correct my poor English. This is really great of you and I'm glad you like my package. Thank you, Fabrizio

**Comment by ftognetto** [comment]
I commit over your pr to explain better a couple of main concepts. If you think it's ok I will proceed to merge the final pr and publish it.
Thanks

**Comment by h-unterp** [comment]
Yes thank you! :-) , but I made 14 commits in this PR, I believe you only included 1 of them. They are all meant to be included. Thank you.

**Comment by ftognetto** [comment]
Yes of course I saw all of them, I only edited one because the other looks good to me :)

**Pull request merged**